spring-boot/admin/src/main/java/eu/coatrack/admin/service/OAuthUserDetailsService.java

```diff
@@ -0,0 +1,106 @@
+package eu.coatrack.admin.service;
+
+/*-
+ * #%L
+ * coatrack-admin
+ * %%
+ * Copyright (C) 2013 - 2022 Corizon | Institut für angewandte Systemtechnik Bremen GmbH (ATB)
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import eu.coatrack.config.github.GithubEmail;
+import lombok.AllArgsConstructor;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClient;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class OAuthUserDetailsService {
+
+    private static final String GITHUB_API_EMAIL = "https://api.github.com/user/emails";
```

Comment:
Why? So from what I understood, we would only have profiles for specific parts of the code which could be different/changed between those profiles, however the URL for the Github API will be always the same.
Regarding the name of the String, I already refactored renamed it in the code (so I applied your suggestion).

Reply:
>  the URL for the Github API will be always the same

We cannot know that, Github may change it in the future. 

Besides, we discussed about supporting other Oauth providers in the future, also for that purpose it may make sense to make it configurable.

Reply:
Done, now it is possible to get the endpoint from the attribute userInfoUri in the application.yml, so no more hardcoded urls :)